drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -44,46 +44,47 @@ end
 --- and supports the OnOff cluster. This is done to bypass the
 --- BRIDGED_NODE_DEVICE_TYPE on bridged devices
 local function find_default_endpoint(device, component)
-  local res = device.MATTER_DEFAULT_ENDPOINT
   local eps = device:get_endpoints(clusters.OnOff.ID)
   table.sort(eps)
   for _, v in ipairs(eps) do
     if v ~= 0 then --0 is the matter RootNode endpoint
-      res = v
-      break
+      return v
     end
   end
   device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
-  return res
+  return device.MATTER_DEFAULT_ENDPOINT
 end
 
-local function initialize_switch(device)
+local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   table.sort(switch_eps)
-  local component_map = {}
 
-  -- For switch devices, the profile components follow the naming convention "switch%d",
-  -- with the exception of "main" being the first component. Each component will then map
-  -- to the next lowest endpoint that hasn't been mapped yet.
-  -- Additionally, since we do not support bindings at the moment, we only want to count
-  -- On/Off clusters that have been implemented as server. This can be removed when we have
+  -- Since we do not support bindings at the moment, we only want to count On/Off
+  -- clusters that have been implemented as server. This can be removed when we have
   -- support for bindings.
   local num_server_eps = 0
+  local main_endpoint = find_default_endpoint(device)
   for _, ep in ipairs(switch_eps) do
     if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-      num_server_eps = num_server_eps + 1;
-      if num_server_eps == 1 then
-        component_map["main"] = ep
-      else
-        component_map[string.format("switch%d", num_server_eps)] = ep
+      num_server_eps = num_server_eps + 1
+      if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+        local name = string.format("%s %d", device.label, num_server_eps)
+        driver:try_create_device(
+          {
+            type = "EDGE_CHILD",
+            label = name,
+            profile = "switch-binary",
```

Comment:
At the moment, this only supports multi-switch devices (just as we had before with MCD). However, I think this could be expand on in a future PR by checking the clusters/device type on each endpoint and mapping to the correct profile now that we have the parent-child paradigm in place.

Reply:
I think that's definitely worth doing, but agreed it can be follow up PR